src/video/video.c

```diff
@@ -196,9 +201,29 @@ static inline void set_screen_viewport(void) {
     glViewport(vp_x, vp_y, g_video_state.viewport_w, g_video_state.viewport_h); // This is used for screen shakes.
 }
 
+void video_render_finish_offscreen(void) {
+    object_array_finish(g_video_state.objects);
+
```

Comment:
This is now rendering with palette and remaps that is one frame late.

Reply:
moved the palette and remap update above the call to video_render_finish_offscreen